DigitalLearningSolutions.Data/DataServices/NotificationPreferencesDataService.cs

```diff
@@ -72,5 +76,55 @@ public IEnumerable<NotificationPreference> GetNotificationPreferencesForDelegate
                 new { delegateId }
             );
         }
+
+        public void SetNotificationPreferencesForAdmin(int? adminId, IEnumerable<int> notificationIds)
+        {
+            if (!adminId.HasValue) return;
+
+            using (var transaction = new TransactionScope())
+            {
+                connection.Execute(
+                    @"DELETE FROM NotificationUsers
+                    WHERE AdminUserId = @adminId",
+                    new { adminId }
+                );
+
+                var notificationIdsWithAdminId =
+                    notificationIds.Select(notificationId => new { adminId, notificationId });
+
+                connection.Execute(
+                    @"INSERT INTO NotificationUsers (NotificationId, AdminUserId)
+                    VALUES (@notificationId, @adminId)",
+                    notificationIdsWithAdminId
+                );
+
+                transaction.Complete();

```

Comment:
Does this work? Could you test if nothing gets committed when "Complete" is removed?
https://stackoverflow.com/questions/6860979/dapper-transactionscope

Reply:
It does work, removing it from a function causes the associated tests to fail.

Reading around there seems to be some confusion because e.g. read-only statements can still 'succeed' (retrieve data) even if in an uncommitted transaction, but the Complete() is required to write. The more developed version would Dispose() in a catch or finally block, but I understand that to be what the using{} block converts to when compiled, so adding it myself would be redundant.

My understanding here is based on this question and answers: https://stackoverflow.com/questions/53657509/do-i-need-transactionscope-complete

Reply:
Ah, nice! Thanks for checking! 🙏